src/ploomber_core/telemetry/telemetry.py

```diff
@@ -0,0 +1,521 @@
+        Checks if the user_email is set in the User conf file (config.yaml).
+        returns True/False accordingly.
+        """
+    settings = UserSettings()
+    return settings.user_email
+
+
+def check_version(package_name, version):
+    """
+    The function checks if the user runs the latest version
+    This check will be skipped if the version_check_enabled is set to False
+    If it's not the latest, notifies the user and saves the metadata to conf
+    Alerting every 2 days on stale versions
+    """
+    settings = UserSettings()
+
+    if not settings.version_check_enabled:
+        return
+
+    internal = Internal()
+    now = datetime.datetime.now()
+
+    # Check if we already notified in the last 2 days
+    if internal.last_version_check and (now -
+                                        internal.last_version_check).days < 2:
+        return
+
+    # check latest version (this is an expensive call since it hits pypi.org)
+    # so we only ping the server when it's been 2 days
+    latest = get_latest_version(package_name, version)
+
+    # If latest version, do nothing
+    if version == latest:
+        return
+
+    click.secho(
+        f"There's a new {package_name} version available ({latest}), "
+        f"you're running {version}. To upgrade: "
+        f"pip install {package_name} --upgrade",
+        fg='yellow')
+
+    # Update latest check date
+    internal.last_version_check = now
+
+
+def _get_telemetry_info(package_name, version):
+    """
+    The function checks for the local config and uid files, returns the right
+    values according to the config file (True/False). In addition it checks
+    for first time installation.
+    """
+    # Check if telemetry is enabled, if not skip, else check for uid
+    telemetry_enabled = check_telemetry_enabled()
+
+    # Check latest version
+    check_version(package_name, version)
+
+    if telemetry_enabled:
+        # Check first time install
+        is_install = check_first_time_usage()
+
+        # if not uid, create
+        internal = Internal()
+        return telemetry_enabled, internal.uid, is_install
+    else:
+        return False, '', False
+
+
+def validate_entries(event_id, uid, action, client_time, total_runtime):
+    event_id = validate_inputs.str_param(str(event_id), "event_id")
+    uid = validate_inputs.str_param(uid, "uid")
+    action = validate_inputs.str_param(action, "action")
+    client_time = validate_inputs.str_param(str(client_time), "client_time")
+    elapsed_time = validate_inputs.opt_str_param(str(total_runtime),
+                                                 "elapsed_time")
+    return event_id, uid, action, client_time, elapsed_time
+
+
+def log_api(action, package_name, version, client_time=None,
+            total_runtime=None, metadata=None):
+    """
+    This function logs through an API call, assigns parameters if missing like
+    timestamp, event id and stats information.
+    """
```

Comment:
please add docstring documenting to document new arguments (package_name and version)